spec/unit/puppet/provider/mongodb_user/mongodb_spec.rb

```diff
@@ -6,19 +6,20 @@
 
   let(:raw_users) do
     [
-      { '_id' => 'admin.root', 'user' => 'root', 'db' => 'admin', 'credentials' => { 'MONGODB-CR' => 'pass' }, 'roles' => [ { 'role' => 'role2', 'db' => 'admin' },  { 'role' => 'role1', 'db' => 'admin' } ] }
+      { '_id' => 'admin.root', 'user' => 'root', 'db' => 'admin', 'credentials' => { 'MONGODB-CR' => '24fed24b83566e4d1f033f33f7caa658' }, 'roles' => [ { 'role' => 'role2', 'db' => 'admin' },  { 'role' => 'role1', 'db' => 'admin' } ] }
     ].to_json
   end
 
   let(:parsed_users) { %w(root) }
 
   let(:resource) { Puppet::Type.type(:mongodb_user).new(
-    { :ensure        => :present,
-      :name          => 'new_user',
-      :database      => 'new_database',
-      :password_hash => 'pass',
-      :roles         => ['role1', 'role2'],
-      :provider      => described_class.name
+    { :ensure         => :present,
+      :name           => 'new_user',
+      :database       => 'new_database',
+      :password_plain => 'pass',
+      :password_hash  => '24fed24b83566e4d1f033f33f7caa658', #must correspond to password_plain, or the provider complains
```

Comment:
changed "pass" to "24fed24b83566e4d1f033f33f7caa658" which is actually the hash of "new_user:mongo:pass" so the check on the method "initialize" of the type goes well